src/server/models/eGauge/eGaugeRequestor.js

```diff
@@ -0,0 +1,183 @@
+const Meter = require('../Meter');
+const { meterTimezone } = require('../../services/meterTimezone');
+class EgaugeRequestor {
+
+	/**
+	 * @param {Meter} meter 
+	 */
+	constructor(meter) {
+		const url = new URL(`https://${meter.url}`);
+		this.apiUrl = `${url.origin}/api`; // can we add an ending slash??
+		this.username = url.searchParams.get('username');
+		this.password = url.searchParams.get('password');
+		this.registerName = url.searchParams.get('registerName');
+		this.meter = meter;
+		this.jwt = undefined;
+		this.registerId = undefined;
+		this.requestHeaders = undefined;
+	}
+
+	/**
+	 * Login to the eGauge device via the provided url on the meter.
+	 * @post this.jwt is set to a valid jwt obtained from the eGauge device.
+	 * @returns jwt
+	 */
+	async login() {
+		// Perform login via digest protocol as documented in eGauge JSON web api.
+		const authUrl = `${this.apiUrl}/auth/`;
+		const loginUrl = authUrl + 'login';
+		let rlm;
+		let nnc;
+		try {
+			await axios.get(authUrl);
+			// the request is expected to fail and supply the realm and nnc
+		} catch (error) {
+			({ rlm, nnc } = (error.response.data));
+		}
+
+		const USERNAME = this.username;
+		const PASSWORD = this.password;
+
+		const CNONCE = crypto.randomBytes(64).toString('hex');
+		const HA1 = md5(`${USERNAME}:${rlm}:${PASSWORD}`);
+
+		const HA2 = md5(`${HA1}:${nnc}:${CNONCE}`);
+
+		const res = await axios.post(loginUrl, {
+			'rlm': rlm,
+			'usr': USERNAME,
+			'nnc': nnc,
+			'cnnc': CNONCE,
+			'hash': HA2
+		});
+
+		const { jwt } = res.data;
+		this.jwt = jwt;
+		this.requestHeaders = { Authorization: `Bearer ${jwt}` };
+		return jwt;
+	}
+
+	/**
+	 * Fetches eGauge data within a time interval. If the reading unit is detected to be Power (recorded as W*s), we convert it kWh.
+	 * @param {number | string} startTimestamp unix start timestamp or 'epoch' which is an eGauge keyword for when the device began gathering data.
+	 * @param {number | string} endTimestamp unix end timestamp or 'soh' which is an eGauge keyword for the start of the current hour.
+	 * @param {number} timeStep time step in seconds.
+	 * @param {string} timezoneUse timezone to interpret the readings in since eGauge uses Unix timestamps that depend on the timezone.
+	 * @returns {Promise<[[number, moment, moment]]>} a matrix whose rows consists of the reading value and the end timestamp as a moment.
+	 */
+	async getMeterReadingsBetweenInterval(startTimestamp, endTimestamp, timeStep, timezoneUse) {
+		const res = await this.get(`${this.apiUrl}/register/?rate&format=json&time=${startTimestamp}:${timeStep}:${endTimestamp}&reg=${this.registerId}`);
+		const data = res.data;
+		if(data.error !== undefined){
+			// An error can occur if the time range is improper. For example, if the endTimestamp is
+			// after the startTimestamp.
+			log.error(data.error);
+			return [];
+		}
+		const meterReadings = [];
+		const readingType = data.registers[0].type;
+		
```

Comment:
Was it okay to remove the s parameter?

Reply:
Yes, it's okay to remove the parameter. The 's' parameter was just to tell moment.unix that the input is the number of seconds after the Unix Epoch. I included it for clarity, but since this is the default parameter value it's okay to remove it.